### PR TITLE
Added Capabilities

### PR DESCRIPTION
**Pull request by rabinsahoo**
Modified Capabilities section.

### CODE REVIEW

File: draft-ietf-roll-mopex-cap.xml [hunk 1]
```diff
@@ -38,7 +40,7 @@
 <?rfc subcompact="no" ?>
 <!-- keep one blank line between list items -->
 <!-- end of list of popular I-D processing instructions -->
-<rfc category="std" docName="draft-ietf-roll-mopex-cap-latest" ipr="trust200902">
+<rfc category="std" docName="draft-ietf-roll-mopex-cap-01" ipr="trust200902">
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: @rabinsahoo , -latest is needed for the make to work ... `make submit` will automatically change the -latest to the next version.

File: draft-ietf-roll-mopex-cap.xml [hunk 2]
```diff
@@ -357,24 +381,28 @@
  0                   1                   2                   3
  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
 +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|   CAPType     |J|C|I|. . . . .| CAPInfo(Opt)
+|   CAPType     |G|I|. . . . . .| CAPInfo(Opt)
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: the 'J' and 'C' flags are removed! These flags were added during interim discussions. We cannot unilaterally remove these without WG knowing it. Why were these flags removed?

File: draft-ietf-roll-mopex-cap.xml [hunk 3]
```diff
@@ -404,15 +432,193 @@
                 this document. Capabilities advertised by non-root nodes are
                 strictly a subset of the capabilities advertised by the root.
             </t>
-            <t>
+            <t>
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: Spaces at the end of line are not allowed as per the id-template. The `make` wont work. Please do a `make`.

File: draft-ietf-roll-mopex-cap.xml [hunk 4]
```diff
@@ -404,15 +432,193 @@
...
         </section>
+		<section title="Capabilities">
+			<section title="Projected Route Capability">
+				<t>
+					<xref target="I-D.ietf-roll-dao-projection"/> proposes mechanisms to compute and install traffic 
+                    engineered paths in the RPL network. It enables an RPL Root to install and maintain 
+                    Projected Routes based on requested path metric, within its DODAG, along with a selected set of 
+					nodes that may or may not include self, for a chosen duration. Projected Route Capability will be 
+					used to enable this TE path calculation. PRC will be an optional global capability. 
+                    Any node that does not understand or support the projected rout functions can still act as an LR.
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: rout -> route.

How can a node that does not understand the projected route functions still act as an LR?

File: draft-ietf-roll-mopex-cap.xml [hunk 5]
```diff
@@ -404,15 +432,193 @@
...
+				<section title="Format of Projected Route Capability">
+					<t>
+						<figure align="center" anchor="prcap-tlv" title="Projected Route Capability TLV">
+							<artwork align="center"><![CDATA[
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+| Type=0x01     |G|I|. . . . . .|     CAPLen    | MOP |  Resvd  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       Routing Merric 1        |      Routing Merric 1         |
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: Merric -> Metric

File: draft-ietf-roll-mopex-cap.xml [hunk 6]
```diff
@@ -404,15 +432,193 @@
...
+					</t>
+					<t>
+						Flags: DODAG root MUST set G bit to 1 plus LRs MUST set it to 0. I bit will always be set to 1.						
+					</t>
+					<t>
+						CAPLen: 8-bit unsigned integer, representing the length in octets of the option, not including 
+						the Option Type and Length fields.						
+					</t>
+					<t>
+						MOP: 3-bit filed indicating the mode of operation of projected route capability.
```
Comment (nyrahul) [inline draft-ietf-roll-mopex-cap.xml]: filed -> field

**Review by nyrahul — COMMENTED** [review]
Comments for the PR.

File: draft-ietf-roll-mopex-cap.xml (hunk 1 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: Updated the xml

File: draft-ietf-roll-mopex-cap.xml (hunk 2 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: Purpose J = Join only as leaf if capability not understood.
In the new scheme, we will use the first bit of the Capability Type to indicate if this Capability is MUST or Optional. G bit says if its Global or local capability. If a capability is MUST and Global then any node that doesn't understand the capability can join as leaf node.

Purpose of C = Copy capability to children
With New Scheme, if the Global (G) bit is set then this capability has to be propagated down the DODAG.

File: draft-ietf-roll-mopex-cap.xml (hunk 4 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: The projected route is for TE path, used for P2P based applications. Without this also a Node can act as the LR in LLN for normal downward traffic.

File: draft-ietf-roll-mopex-cap.xml (hunk 5 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: Updated the xml

File: draft-ietf-roll-mopex-cap.xml (hunk 6 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: Updated

File: draft-ietf-roll-mopex-cap.xml (hunk 3 above)
Comment (rabinsahoo) [reply draft-ietf-roll-mopex-cap.xml]: Updated the XML

**Comment by nyrahul** [comment]
@rabinsahoo - the complete file seems to be changed! Is it because of the ^M issue? Better you can close this PR and raise another one only with the changes. Thanks

**Pull request merged**